enhancements/sig-architecture/82-addon-template/examples/addon-template.yaml

```diff
@@ -0,0 +1,87 @@
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: AddOnTemplate
+metadata:
+  name: hello-template
+spec:
+  addonName: hello-template
+  agentManifests:
+    - kind: Deployment
+      apiVersion: apps/v1
+      metadata:
+        name: hello-template-agent
+        namespace: '{{INSTALL_NAMESPACE}}' # built-in INSTALL_NAMESPACE
+        labels:
+          app: hello-template-agent
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app: hello-template-agent
+        template:
+          metadata:
+            labels:
+              app: hello-template-agent
+          spec:
+            serviceAccountName: hello-template-agent-sa
+            containers:
+              - name: helloworld-agent
+                image: quay.io/zhujian/addon-examples:latest
+                imagePullPolicy: Always
+                args:
```

Comment:
IOW, instead of allowing free-form anything, what if we constrain exactly what we want/need. I can see the need/desire for RBAC on the managedcluster as well.

Reply:
in most cases, the manifests of addon are deployment/sa/RBAC. But for some addons, it also include CRDs and related CRs.

Reply:
the addonTemplate is for developers. Developers may want to define addon agents in a more Kubernetes native way, which is in line with the usage habits of most developers, and we don’t know whether other than deployment/sa/RBAC/CRDs/CRs additional resources are needed

Reply:
Those examples help, thanks.